From 44bf00c25d25532d328d0635daaa235699aa624e Mon Sep 17 00:00:00 2001
From: Robin Gong <yibin.gong@nxp.com>
Date: Tue, 7 Nov 2017 17:30:54 +0800
Subject: [PATCH 2814/5242] MLK-16765-5 soc: imx: sc: ipc: enable
 SC_IRQ_BUTTON

commit  1d53dc55f9415c27a916c4bd370814a95a4ecbf5 from
https://source.codeaurora.org/external/imx/linux-imx.git

Enable SC_IRQ_BUTTON interrupt in SCU side.

Signed-off-by: Robin Gong <yibin.gong@nxp.com>
Reviewed-by: Peng Fan <peng.fan@nxp.com>
Reviewed-by: Anson Huang <Anson.Huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/soc/imx/sc/main/ipc.c |    7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/soc/imx/sc/main/ipc.c b/drivers/soc/imx/sc/main/ipc.c
index 7ef0175..10f24d89 100644
--- a/drivers/soc/imx/sc/main/ipc.c
+++ b/drivers/soc/imx/sc/main/ipc.c
@@ -360,6 +360,13 @@ int __init imx8_mu_init(void)
 	if (sciErr)
 		pr_info("Cannot request ALARM_RTC interrupt\n");
 
+	/* Request for the ON/OFF interrupt. */
+	sciErr = sc_irq_enable(mu_ipcHandle, SC_R_MU_0A, SC_IRQ_GROUP_WAKE,
+			       SC_IRQ_BUTTON, true);
+
+	if (sciErr)
+		pr_info("Cannot request ON/OFF interrupt\n");
+
 	pr_info("*****Initialized MU\n");
 	return scu_mu_id;
 }
-- 
1.7.9.5

